File: CS/WebSite/App_Code/SkipedOperation.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

/// <summary>
/// Summary description for SkipedOperation
/// </summary>
public class SkipedOperation {
    private List<int> skippedDetailBands;

    public List<int> SkippedDetailBands {
        get {
            if(HttpContext.Current.Session["SkipedList"] == null) {
                HttpContext.Current.Session["SkipedList"] = new List<int>();
            }
            return HttpContext.Current.Session["SkipedList"] as List<int>;
        }
        set { skippedDetailBands = value; }
    }

    public void UpdateDetail(int detailID) {
        if(SkippedDetailBands.Contains(detailID))
            SkippedDetailBands.Remove(detailID);
        else
            SkippedDetailBands.Add(detailID);
    }
}

File: VB/WebSite/App_Code/SkipedOperation.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web

''' <summary>
''' Summary description for SkipedOperation
''' </summary>
Public Class SkipedOperation
	Private skippedDetailBands_Renamed As List(Of Integer)

	Public Property SkippedDetailBands() As List(Of Integer)
		Get
			If HttpContext.Current.Session("SkipedList") Is Nothing Then
				HttpContext.Current.Session("SkipedList") = New List(Of Integer)()
			End If
			Return TryCast(HttpContext.Current.Session("SkipedList"), List(Of Integer))
		End Get
		Set(ByVal value As List(Of Integer))
			skippedDetailBands_Renamed = value
		End Set
	End Property

	Public Sub UpdateDetail(ByVal detailID As Integer)
		If SkippedDetailBands.Contains(detailID) Then
			SkippedDetailBands.Remove(detailID)
		Else
			SkippedDetailBands.Add(detailID)
		End If
	End Sub
End Class